homeassistant/components/sensibo/climate.py

```diff
@@ -141,44 +144,64 @@ async def async_assume_state(service):
 class SensiboClimate(ClimateEntity):
     """Representation of a Sensibo device."""
 
-    def __init__(self, client, data, units):
-        """Build SensiboClimate.
-
-        client: aiohttp session.
-        data: initially-fetched data.
-        """
+    def __init__(self, client: SensiboClient, data: dict[str, Any], units: str) -> None:
+        """Initiate SensiboClimate."""
         self._client = client
         self._id = data["id"]
         self._external_state = None
         self._units = units
-        self._available = False
-        self._do_update(data)
         self._failed_update = False
+        self._attr_available = False
+        self._attr_unique_id = self._id
+        self._attr_temperature_unit = (
+            TEMP_CELSIUS if data["temperatureUnit"] == "C" else TEMP_FAHRENHEIT
+        )
+        self._do_update(data)
+        self._attr_target_temperature_step = (
+            1 if self.temperature_unit == units else None
+        )
         self._attr_device_info = DeviceInfo(
             identifiers={(DOMAIN, self._id)},
-            name=self._name,
+            name=self._attr_name,
             manufacturer="Sensibo",
             configuration_url="https://home.sensibo.com/",
             model=data["productModel"],
             sw_version=data["firmwareVersion"],
             hw_version=data["firmwareType"],
-            suggested_area=self._name,
+            suggested_area=self._attr_name,
         )
 
-    @property
-    def supported_features(self):
-        """Return the list of supported features."""
-        return self._supported_features
-
-    def _do_update(self, data):
-        self._name = data["room"]["name"]
-        self._measurements = data["measurements"]
+    def _do_update(self, data) -> None:
+        self._attr_name = data["room"]["name"]
         self._ac_states = data["acState"]
-        self._available = data["connectionStatus"]["isAlive"]
+        self._attr_extra_state_attributes = {
+            "battery": data["measurements"].get("batteryVoltage")
+        }
+        self._attr_current_temperature = convert_temperature(
+            data["measurements"].get("temperature"),
+            TEMP_CELSIUS,
+            self._attr_temperature_unit,
+        )
+        self._attr_current_humidity = data["measurements"].get("humidity")
+
+        self._attr_target_temperature = self._ac_states.get("targetTemperature")
+        self._attr_hvac_mode = (
+            HVAC_MODE_OFF
+            if not self._ac_states["on"]
+            else SENSIBO_TO_HA.get(self._ac_states["mode"], "")
+        )
```

Comment:
If a teranry spans multiple lines, our standard is to convert it to an `if` block.